connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connect.service
+
+import java.util.UUID
+import java.util.concurrent.{ConcurrentHashMap, ConcurrentMap}
+
+import collection.JavaConverters._
+
+import org.apache.spark.connect.proto
+import org.apache.spark.sql.SparkSession
+
+/**
+ * Object used to hold the Spark Connect session state.
+ */
+case class SessionHolder(
+  userId: String,
+  sessionId: String,
+  session: SparkSession) {
+
+  val executePlanOperations: ConcurrentMap[String, ExecutePlanHolder] =
```

Comment:
I am wondering if we need this kind of mapping for the purpose of supporting `interrupt all`. This map and those code around it seems to support `interrupt one`? 

If to call all, then just cancel through every of the execution plan holder?

Reply:
It's not strictly needed at this very moment, but this is a foundation needed for the immediate followup work of actually detaching the execution from the RPC handler described in the PR description. This is then needed both for future interrupt one, and 
I'd like to set this foundation to make it easier for followup work without running into conflicting changes.

Reply:
Sound good if this is for the follow-ups.